src/fstests/runner.py

```diff
@@ -67,60 +78,52 @@ def _run_single_job(self, tarball_node, device):
         finally:
             return True
 
-    def check_kvm_xfstest_env(self):
+    def run(self, args):
+        sub_id = None
```

Comment:
This seems more complicated than it needs to be, but I guess it works.

It could simply be (without exception handling to make it easier to read):
```python
if args.node_id:
  tarball_node = self._db.get_node(args.node_id)
  self._run_single_job(tarball_node, device)
else:
  sub_id = self._db.subscribe_node_channel()
  while True:
    tarball_node = self._db.receive_node(sub_id)
    self._run_single_job(tarball_node, device)
  self._db.unsubscribe(sub_id)
```

Reply:
Erm, I didn't mean you should drop exception handling.  Just for my code snippet it was easier to read without it.

Reply:
ops, I misunderstood that, now it is updated.